journey-builder-triggers/template.ts

```diff
@@ -0,0 +1,161 @@
+import { RecommendationsConfig, RecipeReferenceLookup, RecipeReference, recommend } from "recs";
+import { UserAttributeLookup, UserAttributeReference, UserSegmentLookup, UserSegmentReference } from "common";
+
+export class JourneyBuilderTriggersUtils {
+    static defaultCatalogObject = "Product";
+    static defaultCatalogObjectAttributes = ["id", "name", "imageUrl", "url", "price"];
+    static maxRecommendations = 8;
+
+    // ================================== Catalog Localization =====================================
+    // If you use catalog localization and would like to receive localized catalog data
+    // in Marketing Cloud, change the value of 'isCatalogLocalized' to true
+    // =============================================================================================
+    static isCatalogLocalized = false;
+
+    static getFlatItems(context: CampaignComponentContext, catalogObject: string, ids: string[]): any[] {
+        const locale: string = context.locale || "";
+
+        return context.services.catalog
+            .findItems(catalogObject, ids)
+            .map(item => item.toFlatJSON(JourneyBuilderTriggersUtils.defaultCatalogObjectAttributes, JourneyBuilderTriggersUtils.isCatalogLocalized ? locale : null));
```

Comment:
we have a standard we follow for global templates where we break long lines to under 120 characters to improve readability
```suggestion
            .map(item => item.toFlatJSON(
                JourneyBuilderTriggersUtils.defaultCatalogObjectAttributes, 
                JourneyBuilderTriggersUtils.isCatalogLocalized ? locale : null
            ));
```

Reply:
@nriserEG didn't we also make a call about always including parens for arguments in callbacks? Like
```js
.map((item) => item.toFlatJSON(
```
instead of
```js
.map(item => item.toFlatJSON(
```

_Aside: we should probably write some of these standards down somewhere or add a linter to enforce them 😅 _

Reply:
yeah I think you're right